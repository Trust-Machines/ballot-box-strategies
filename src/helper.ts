import {NETWORK} from './types';
import {NETWORKS} from './constants';
import strategies, {runStrategy} from './index';

export const getNetworkByName = (name: NETWORK) => {
    return NETWORKS[name];
}

export const testStrategy = async (s: string) => {
    const strategy = strategies[s];
    const {testConfig} = strategy;

    // Symbol definition must be exists in the schema
    expect(strategy.schema.symbol === undefined).toEqual(false);

    for (let address of testConfig.addresses) {
        const resp = await runStrategy(s, {
            network: getNetworkByName(testConfig.network),
            address,
            blockTip: testConfig.blockTip,
            blockHeight: testConfig.blockHeight,
            options: testConfig.options
        });
        expect(resp).toMatchSnapshot(`${s}-${address}`);
    }
}
