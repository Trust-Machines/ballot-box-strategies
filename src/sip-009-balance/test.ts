import {testStrategy} from '../helper';

it('sip-009-balance', async () => {
    await testStrategy('sip-009-balance');
});
