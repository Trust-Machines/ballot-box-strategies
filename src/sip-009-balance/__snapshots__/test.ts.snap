// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sip-009-balance: sip-009-balance-SP2DXHX9Q844EBT80DYJXFWXJKCJ5FFAX50CQQAWN 1`] = `1`;

exports[`sip-009-balance: sip-009-balance-SP2N7SK0W83NJSZHFH8HH31ZT3DXJG7NFE5VYT9SJ 1`] = `0`;

exports[`sip-009-balance: sip-009-balance-SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7 1`] = `0`;
